### PR TITLE
Add tag operations for sales

### PR DESCRIPTION
**Pull request by Asthenosphere**
Closes #96 
Result of tag edit for sales:
Screenshot 2020-10-13 at 5 05 01 PM

Result of tag delete for sales:
Screenshot 2020-10-13 at 5 05 21 PM

Result of adding a new sale with a new tag:
Screenshot 2020-10-13 at 5 07 07 PM

### CODE REVIEW

File: src/main/java/seedu/address/model/AddressBook.java [hunk 1]
```diff
@@ -236,9 +249,19 @@ public int findByContactTag(Tag target) {
     /**
      * Returns the number of sale items that are associated with the {@code target} tag.
```
Comment (sebastiantoh) [inline src/main/java/seedu/address/model/AddressBook.java]:
```suggestion
     * Lists all sale items associated with {@code target} tag.
```

**Review by sebastiantoh — APPROVED** [review]
LGTM!

**Comment by sebastiantoh** [comment]
Not really relevant to this PR, but I think the parameter and example for editing a tag is missing the updated tag name. Example should be something like `tag edit 1 t/furniture`. Feel free to merge this PR if you want to resolve this in another PR.

image

File: src/main/java/seedu/address/logic/commands/tag/EditCommand.java [hunk 2]
```diff
@@ -25,7 +25,7 @@
             + "Note that all contacts or sales associated with this tag "
             + "will be updated automatically with the updated tag.\n"
             + "Parameters: INDEX (must be a positive integer)\n"
```
Comment (sebastiantoh) [inline src/main/java/seedu/address/logic/commands/tag/EditCommand.java]: ps still got the parameter here that needs to be updated as well

Comment (Asthenosphere) [reply src/main/java/seedu/address/logic/commands/tag/EditCommand.java]: Right sorry haha

**Pull request merged**